src/pages/Transfer/TransferForm/index.tsx

```diff
@@ -0,0 +1,220 @@
+
+import * as React from 'react';
+import {
+  useSelector,
+  useDispatch
+} from 'react-redux';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import clsx from 'clsx';
+import { toast } from 'react-toastify';
+import { newMonetaryAmount, CurrencyUnit } from '@interlay/interbtc-api';
+import { Currency } from '@interlay/monetary-js';
+
+import TokenAmountField from '../TokenAmountField';
+import {
+  Tokens,
+  TokenOption
+} from 'components/Tokens';
+import SubmitButton from 'components/SubmitButton';
+import FormTitle from 'components/FormTitle';
+import TextField from 'components/TextField';
+import ErrorModal from 'components/ErrorModal';
+import {
+  ParachainStatus,
+  StoreType
+} from 'common/types/util.types';
+import { showAccountModalAction } from 'common/actions/general.actions';
+import isValidPolkadotAddress from 'utils/validation/validate-address';
+import STATUSES from 'utils/constants/statuses';
+import {
+  KUSAMA,
+  POLKADOT
+} from 'utils/constants/relay-chain-names';
+
+const TRANSFER_AMOUNT = 'wrapped-token-input-amount';
+const RECIPIENT_ADDRESS = 'collateral-token-address';
+
+type TransferFormData = {
+  [TRANSFER_AMOUNT]: string;
+  [RECIPIENT_ADDRESS]: string;
+}
+
+const TransferForm = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+
+  const {
+    parachainStatus,
+    address
+  } = useSelector((state: StoreType) => state.general);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset
+  } = useForm<TransferFormData>({
+    mode: 'onChange'
+  });
+
+  const [activeToken, setActiveToken] = React.useState<TokenOption | undefined>(undefined);
+  const [accountSet, setAccountSet] = React.useState<boolean | undefined>(undefined);
+  const [submitStatus, setSubmitStatus] = React.useState(STATUSES.IDLE);
```

Comment:
@tomjeatt 
Can we use react-query's mutation instead of manually handling this?

Reply:
Do you mean the submitStatus? Possibly yes, but I've just followed what we've done with the other forms (I haven't actually changed this, it's as it was in the original transfer form). Might be a nice improvement across the app but I don't think it's critical as long as the the transfer form is consistent with the other forms.

Reply:
I myself implemented all of the forms but at that time, the react-query was not in place unfortunately. I'd say it would be good to use react-query's mutation for new updates like this because it's better for DX as it's more declarative. OK then.

Reply:
I agree with that - out of scope for this ticket, but something we can add to the backlog as an improvement as and when we add new features/refactor existing ones.